clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/PatchExecutor.scala

```diff
@@ -149,5 +142,29 @@ class PatchExecutor[CI <: ClioIndex](patchCommand: PatchCommand[CI]) extends Exe
       )
     )
   }
+}
+
+object PatchExecutor {
 
+  /**
+    * Given a JSON object containing metadata fields which should be "patched" into
+    * Clio documents, build a JSON object containing an Elasticsearch query which
+    * will return only documents with at least one unset value amongst the fields-to-patch.
+    *
+    * Used to avoid pulling documents from Clio that are guaranteed to not be re-
+    * upserted, since patching doesn't overwrite existing values.
```

Comment:
Do we handle the case were the before/after data is the same in `mergeMetadata` above?

Reply:
It was handled in the `for` comprehension, but I've shuffled things around to make it happen in `mergeMetadata`